src/extensions/emulator/optionsHelper.ts

```diff
@@ -46,6 +43,27 @@ export async function buildOptions(options: any): Promise<any> {
   return options;
 }
 
+// Exported for testing
+export async function getParams(options: any, extensionSpec: ExtensionSpec) {
+  const projectId = getProjectId(options, false);
+  const userParams = await paramHelper.readParamsFile(options.testParams);
+  const autoParams = {
+    PROJECT_ID: projectId,
+    EXT_INSTANCE_ID: extensionSpec.name,
+    DATABASE_INSTANCE: projectId,
+    DATABASE_URL: `https://${projectId}.firebaseio.com`,
+    STORAGE_BUCKET: `${projectId}.appspot.com`,
+  };
+  const unsubbedParamsWithoutDefaults = Object.assign(autoParams, userParams);
+
+  const unsubbedParams = extensionsHelper.populateDefaultParams(
+    unsubbedParamsWithoutDefaults,
+    extensionSpec.params
+  );
+  // Run a substitution to support  params that reference auto populated params.
+  return extensionsHelper.substituteParams<Record<string, string>>(unsubbedParams, unsubbedParams);
```

Comment:
nit*: I think this it would be slightly more clear if the arguments were `(unsubbedParams, autoParams)`

Reply:
Addressed below, but this is intentional to cover cases where user params are referenced. I'll clarify this comment